src/components/context/product-context.js

```diff
@@ -0,0 +1,25 @@
+import { createContext, useContext, useReducer } from "react";
+import { productReducer } from "../reducer/productReducer";
+
+const initialState = {
+  sortByPrice: null,
+  sortByRating: null,
+  categories: {
+    Anjunadeep: false,
+    Anjunabeats: false,
+    Above_And_Beyond: false,
+  },
+};
+
+const ProductContext = createContext(null);
+const useProduct = () => useContext(ProductContext);
+const ProductProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(productReducer, initialState);
```

Comment:
more specific names here would help if there are more reducers . the generic names can cause confusion